python_modules/dagster/dagster_tests/execution_tests/engine_tests/test_step_delegating_executor.py

```diff
@@ -384,7 +384,7 @@ def test_execute_using_repository_data():
                 {"compute_cacheable_data_called", "get_definitions_called"}
             )
             assert call_counts.get("compute_cacheable_data_called") == "1"
-            assert call_counts.get("get_definitions_called") == "5"
+            assert call_counts.get("get_definitions_called") == "6"
```

Comment:
specifically this codepath is the reason for the additional get_definitions_called https://github.com/dagster-io/dagster/blob/586230ddc1e642ed6f7138e167a6e64ac815fcc6/python_modules/dagster/dagster/_core/execution/api.py#L224-L240 

We appear to be already incurring that extra load for all AMP and asset backfill jobs though

Reply:
ultimately i do not actually care if the asset selection is used during execution at all - i just want to be able to efficiently fetch the set of assets that were planned to be materialized by this asset after the fact, without scanning the event log